bundles/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java

```diff
@@ -1,499 +1,494 @@
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            boolean inMessage = false;
+            int syncBytes = 0;
+
+            while (true) {
+                // if timed out, exit
+                int c = is.read();
+                if (c < 0) {
+                    return null;
+                }
+
+                byte b = (byte) c;
+
+                if (b == FRAME_SYNC) {
+                    if (inMessage) {
+                        if (syncBytes == 0) {
+                            // special sequence or end of message
+                            // wait for next byte
+                        } else {
+                            logger.warn("Received frame sync bytes, discarding input: {}", baos.size());
+                            // clear gathered bytes, we wait for new message
+                            inMessage = false;
+                            baos.reset();
+                        }
+                    }
+                    ++syncBytes;
+                } else {
+                    if (inMessage) {
+                        if (syncBytes == 0) {
+                            // in sync, we have next message byte
+                            baos.write(b);
+                        } else if (syncBytes == 1) {
+                            if (b == FRAME_SYNC_ESC) {
+                                baos.write(FRAME_SYNC);
+                            } else if (b == FRAME_END[1]) {
+                                // end of message
+                                break;
+                            } else {
+                                logger.warn("Received invalid byte {}, discarding input: {}", String.format("%02X", b),
+                                        baos.size());
+                                // clear gathered bytes, we have new message
+                                inMessage = false;
+                                baos.reset();
+                            }
+                        } else {
+                            logger.error("Sync bytes in message: {}", syncBytes);
+                        }
+                    } else if (syncBytes >= 2) {
+                        // synced, we have first message byte
+                        inMessage = true;
+                        baos.write(b);
+                    } else {
+                        // discard all bytes until synced
+                    }
+                    syncBytes = 0;
+                }
+
+                // if meanwhile thread has been interrupted, exit the loop
+                if (Thread.interrupted()) {
+                    return null;
+                }
+            }
+
+            // return read message
+            return SatelMessage.fromBytes(baos.toByteArray());
+
+        } catch (IOException e) {
+            if (!Thread.interrupted()) {
+                logger.error("Unexpected exception occurred during reading a message", e);
+            }
```

Comment:
"other IOExceptions" meant when the thread was interrupted, so I think you're saying that only when the IOException is not the result of an interrupted thread it needs to be logged.  If so, fair enough!